airflow/providers/amazon/aws/sensors/glue.py

```diff
@@ -61,10 +62,13 @@ def __init__(
         self.errored_states: list[str] = ["FAILED", "STOPPED", "TIMEOUT"]
         self.next_log_token: str | None = None
 
+    @cached_property
+    def hook(self):
+        return GlueJobHook(aws_conn_id=self.aws_conn_id)
```

Comment:
Some of these are passing region and others are not.  Is that intentional or should we standardize that as well?

Reply:
I mostly kept the existing behavior. I don't know what is a real use case for passing the region, maybe that depends on the operator/sensor ?

Reply:
Somewhere in the past I have an idea to keep Sensors/Operators more consistent and provide all of supported parameters in Hook.
I think it safe to pass `region_name` because we set it to None by default in Operators/Sensors.

Reply:
BTW , it is not related to this PR, but we have a 3 different hooks which is wraps around Glue.Client 🤦 
- airflow.providers.amazon.aws.hooks.glue.GlueJobHook
- airflow.providers.amazon.aws.hooks.glue_catalog.GlueCatalogHook
- airflow.providers.amazon.aws.hooks.glue_crawler.GlueCrawlerHook

Reply:
> BTW , it is not related to this PR, but we have a 3 different hooks which is wraps around Glue.Client facepalm

Yeah, there is also a BatchWaiterHook which is an oddball too.  Some Day :tm:

Reply:
In one recent PR I mark Hooks as Thin and Thick wrapper, so we could easily find them in "Some Day" 

```console

❯ grep -rl 'thick wrapper' airflow/providers/amazon/aws/hooks
airflow/providers/amazon/aws/hooks/emr.py
airflow/providers/amazon/aws/hooks/dynamodb.py
airflow/providers/amazon/aws/hooks/batch_client.py
airflow/providers/amazon/aws/hooks/kinesis.py
airflow/providers/amazon/aws/hooks/glue.py
airflow/providers/amazon/aws/hooks/sagemaker.py
airflow/providers/amazon/aws/hooks/ec2.py
airflow/providers/amazon/aws/hooks/datasync.py
airflow/providers/amazon/aws/hooks/athena.py
airflow/providers/amazon/aws/hooks/s3.py
airflow/providers/amazon/aws/hooks/elasticache_replication_group.py

```

Reply:
Tack onto the "Some Day" list, converting as many hooks as possible to thin hooks, maybe with a Protocol class for IDE type hinting and completion.  I didn't like the EcsProtocol and BatchProtocol classes at first, but they grew on me.

Anyway, we should maybe get a list of "easy but tedious suggestions" like this for new contributors to flip through, but we;'re getting way off track here :stuck_out_tongue:

Reply:
I also like an idea @o-nikolas generate hooks for all supported services https://github.com/apache/airflow/discussions/28560#discussioncomment-4585430